README.md

```diff
@@ -264,3 +264,103 @@ nomad job status virt-example
 
 virsh list
 ```
+## Debugging a VM
+
+Sometimes things dont go as plan and extra tools are necessary to find the problem.
+Here are some strategies to debug a failing vm:
+
+### Connecting to a VM
+
+By default, cloud images are password protected, by adding a `default_user_password`
+a new password is assigned to the default user of the used distribution: `ubuntu` for ubuntu
+`fedora` for fedora, `root` for alpine... 
+By running `virsh console [vm-name]`, a terminal is started inside the vm that will allow an internal inspection of the vm.
+
+```
+$ virsh list
+ Id   Name                 State
+------------------------------------
+ 1    virt-task-8bc0a63f   running
+
+$ virsh console virt-task-8bc0a63f 
+Connected to domain 'virt-task-8bc0a63f'
+Escape character is ^] (Ctrl + ])
+
+nomad-virt-task-8bc0a63f login: ubuntu
+Password:
+```
+
+If no login promt shows up, it can mean the virtual machine is not booting, adding some extra 
+space to the disk can solve the problem, remember the disk has to fit the cloud image
+plus any other process running in the vm.
```

Comment:
```suggestion
If no login prompt shows up, it can mean the virtual machine is not booting and 
adding  some extra  space to the disk may solve the problem. Remember the disk 
has to fit the root image plus any other process running in the VM.
```

What does "plus any other process" refer to here?

Reply:
Whatever you might install inside de vm

Reply:
Isn't that bound by the size of the image's file system? Ex. if `/` is 10GiB, we need to fit 10GiB of space but anything installed during boot is going to be inside that same 10GiB.